internal/controller/apiextensions/claim/configurator_test.go

```diff
@@ -270,6 +271,90 @@ func TestCompositeConfigure(t *testing.T) {
 				},
 			},
 		},
+		"NameMismatchExistingXR": {
+			reason: "If there is an external-name mismatch between a claim and a statically provisioned composite resource it's bound to, an error is expected",
+			args: args{
+				cm: &claim.Unstructured{
+					Unstructured: unstructured.Unstructured{
+						Object: map[string]interface{}{
+							"metadata": map[string]interface{}{
+								"namespace": ns,
+								"name":      name,
+								"annotations": map[string]interface{}{
+									// This should be reset to the equivalent
+									// composite resource value, since it has
+									// most likely already taken effect and
+									// cannot be updated retroactively.
```

Comment:
Is this comment true? I think we'll return an error rather than resetting this annotation.

Reply:
Copy/paste error. Thanks for pointing it out. I've rewritten it to reflect current expectations.